test/runtests.jl

```diff
@@ -92,18 +92,60 @@ up_path = joinpath(@__DIR__, "UsesPreferences")
             using UsesPreferences, Test, Preferences
             using Base: UUID
             @test load_preference($(repr(up_uuid)), "username") === nothing
+            # @test !UsesPreferences.has_username()
             @test UsesPreferences.get_username() === nothing
             UsesPreferences.set_username("giordano")
             @test UsesPreferences.get_username() == "giordano"
         """)
 
         # This does not cause a recompilation, and we can also get the username back again:
         username_test = """
-        using UsesPreferences, Test
+        using UsesPreferences, Test, Preferences
         @test UsesPreferences.get_username() == "giordano"
         """
         output = activate_and_run(up_path, username_test; env=Dict("JULIA_DEBUG" => "loading"))
         @test !did_precompile(output)
+
+        snippets = [
+            "UsesPreferences.delete_username()",
+            """
+            !@test UsesPreferences.has_username()
+            UsesPreferences.set_username("giordano")
+            @test UsesPreferences.get_username() == "giordano"
+            """,
+            "UsesPreferences.delete_username()",
+            """
+            !@test UsesPreferences.has_username()
+            UsesPreferences.set_username("giordano")
+            @test UsesPreferences.get_username() == "giordano"
+            """,
+            """
+            delete_preferences!($(repr(up_uuid)), "username"; block_inheritance = false)
+            """,
+            """
+            !@test UsesPreferences.has_username()
+            UsesPreferences.set_username("giordano")
+            @test UsesPreferences.get_username() == "giordano"
+            """,
+            """
+            delete_preferences!($(repr(up_uuid)), "username"; block_inheritance = true)
+            """,
+            """
+            !@test UsesPreferences.has_username()
+            UsesPreferences.set_username("giordano")
+            @test UsesPreferences.get_username() == "giordano"
+            """,
```

Comment:
It seems to me that this same chunk of code is repeated a bunch of times.  Could you save these to variables with descriptive verb-like names (like `delete_all_usernames`, `set_and_check`, etc...) and then arrange them so we can more easily see the flow of actions that you're testing?

Reply:
Done.